fees/makina/index.ts

```diff
@@ -1,68 +1,132 @@
 import { Adapter, FetchOptions, FetchResultV2 } from "../../adapters/types";
 import { CHAIN } from "../../helpers/chains";
+import { addTokensReceived } from "../../helpers/token";
 import ADDRESSES from "../../helpers/coreAssets.json";
+import { METRIC } from "../../helpers/metrics";
 
 const ABI = {
   FeesMinted: "event FeesMinted(uint256 amount)",
+  TotalAumUpdate: "event TotalAumUpdated(uint256 totalAum)",
   getSharePrice: "function getSharePrice() view returns (uint256)",
-  decimals: "uint8:decimals",
-  perfFeeRate: "function perfFeeRate() view returns (uint256)",
-  perfFeeSplitBps: "function perfFeeSplitBps() view returns (uint256[])",
-  sharePriceWatermark: "function sharePriceWatermark() view returns (uint256)",
   totalSupply: "function totalSupply() view returns (uint256)",
+  decimals: "uint8:decimals",
 };
 
 type MachineConfig = {
   shareToken: string;
   accountingToken: string;
-  depositor: string;
-  preDepositVault?: string;
-  sharePriceOracle: string;
-  feeManager: string;
   machine: string;
+  protocolReceiver: string; // where protocol revenue is sent, as we cannot depend on the order to be consistent
 };
 
+// modified from helpers/erc4626.ts
+async function getMachineYield({
+  options,
+  shareTokens,
+  machines,
```

Comment:
this function is always used with ONE machine at at a time, i suggest to conver t this to a string instead of string[]


```suggestion
  machine,
```